source/routing/tests/raptor_test.cpp

```diff
@@ -2793,3 +2793,98 @@ BOOST_AUTO_TEST_CASE(reader_with_invalid_vj_extensions) {
     BOOST_CHECK_EQUAL(res.at(0).items.size(), 5);
 }
 
+/*
+  |-------------|                                               |--------------------------|
+  | StopArea:A  |           vj:A                                |      StopArea:B          |
+  |-------------|============================================== |--------------------------|
+  | StopPoint:A |                                               |                          |
+  |-------------|                                               |    StopPoint:B0          |
+                                                                |          ||              |
+                                                                |          ||              |
+                                                                |          ||  stay_in     |
+                                                                |          ||              |
+                                                                |          ||              |
+                                                                |    StopPoint:B1          |
+                                                                |          ||              |
+                                                                |--------------------------|
+                                                                           ||
+                                                                           ||
+                                                                           ||
+  |-------------|                                                          ||
+  | StopArea:C  |                         vj:B                             ||
+  |-------------|============================================================
+  | StopPoint:C |
+  |-------------|
+
+We must have the same answer in the two cases: clockwise = true or false
+
+Before correction if clockwise = true, found in the response StopPoint:B0 >> StopPoint:B1 section
+
+See : http://jira.canaltp.fr/browse/NAVITIAII-2062
+
+*/
+
+BOOST_AUTO_TEST_CASE(fix_datetime_represents_arrival_departure) {
+    ed::builder b("20120614");
+
+    b.sa("stop_area:A", 0, 0, false)("stop_point:A");
+    b.sa("stop_area:B", 0, 0, false)("stop_point:B1")("stop_point:B0");
+    b.sa("stop_area:C", 0, 0, false)("stop_point:C");
+
+    b.vj("A", "1111111", "block1", true, "vj:A")("stop_point:A", "8:05"_t, "8:05"_t)
+            ("stop_point:B0", "08:40"_t, "08:40"_t);
+    b.vj("A", "1111111", "block1", true, "vj:B")("stop_point:B1", "08:40"_t, "09:05"_t)
```

Comment:
minor:

you can use the new composite setter:

`b.vj("A").name("vj:A").block("block1")("stop_point:A", "8:05"_t, "8:05"_t)("stop_point:B0", "08:40"_t, "08:40"_t);`

(but I think if you do so, you'll need to add the `block` setter)